### DISCUSSION TITLE
New begginer with Babylon.js + React

### DISCUSSION THREAD

**Discussion by map1336** [brianzinn/react-babylonjs#143] [Q&A]
Hi! guys, i'm currently learning BabylonJS, and saw in Babylon.js docs recommended this library.
this is an amazing concept. I saw the stories(Examples) and they look pretty cool!, but sadly i'm going to a tough time learning to use this library, with JSX in mind, 

**Using JSX**
im trying to implement createGroundFromHeightMap, but i can't find any example in how to implement it
how to implement reflections

**Resources used to learn**
https://brianzinn.github.io/create-react-app-babylonjs/
https://github.com/brianzinn/react-babylonjs/tree/master/stories/babylonjs

1-Is this project gonna have any update in the docs soon?
 even tough Babylon.js api provides docs by default, i do not know how to use their respective functions with this framework

2-I think there's gotta be a good tutorial about making a game with this amazing library, that would be a perfect head start for us the rookies 😄

**Comment by brianzinn** [comment]
Great question. I have no examples, but you should be able to with:
`<groundFromHeightMap url={“...”} />`

you can get all the options from here:

https://doc.babylonjs.com/typedoc/classes/babylon.meshbuilder#creategroundfromheightmap

the options are immediate properties of what you build. I’m not at a computer right now to check, but that should work.

Good idea to make a tutorial for building a game!

**Reply by map1336** [reply]
Yes, babylon currently is offering one
https://doc.babylonjs.com/guidedLearning/createAGame

So making a game, with this library, will be so awesome!

**Reply by brianzinn** [reply]
hi @ipr0310 - that's a good idea.  It would be fun to port that over and see where a declarative markup and React composition would benefit.  Also to compare where the pure imperative coding is better.  I may start by porting this old babylonjs 2.5 project to CRA (and fixing the mesh clicking first).
https://github.com/brianzinn/react-redux-babylonjs-starter-kit

Then from there to implement that game.  I already found some useful snippets in there like using EngineFactory.  It will be a good proving ground - I don't expect to have spare time though till September unfortunately...

**Reply by map1336** [reply]
Hi! i know it has been a long, long while 😀😀😀 are there any news about a type of tutorial in the roadmap?

**Reply by brianzinn** [reply]
The only thing on the roadmap right now is a V4 that supports tree-shaking and more of babylon v5 API (while remaining backwards compatible if possible with 4.2).  actually there isn't a roadmap, but I still think it would be a great idea to port that game over!!

It's going to be a lot of work to get tree-shaking working actually for v4, as I need to split this library into 3 parts and there will be lots of documentation on the new registration methods.

I do appreciate your follow-up.  If you wanted to start porting the game I would be happy to contribute - I don't know if you maybe already got something going with a game using React?  I built my first game using React + Babylon - it was a turn-based game and was a good learning experience.
